src/world/Explosion.php

```diff
@@ -195,7 +195,7 @@ public function explodeB() : bool{
 				}
 
 				$entity->attack($ev);
-				$entity->setMotion($motion->multiply($impact));
+				$entity->addMotion($motion->x * $impact, $motion->y * $impact, $motion->z * $impact);
```

Comment:
`addMotion` requires the `x`, `y`, and `z` arguments, also that way will create an object unnecessarily.

Reply:
> The math logic can also be written as `$entity->getMotion()->addVector($motion->multiply($impact))` for a clearer understanding of what is happening.

No, because `Entity::getMotion()` returns a clone of the original object.

Reply:
setMotion should be used here, because addMotion doesn't call EntityMotionEvent. Say what you want about EntityMotionEvent, but the behaviour shouldn't be changing if we can help it.